firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/Endian.h

```diff
@@ -14,6 +14,23 @@
 
 namespace Pufferfish::Util {
 
+constexpr void read_ntoh(const uint8_t *network_endian, int16_t &host_endian) noexcept {
```

Comment:
For the `int16_t` and `int64_t` implementations, did you copy-paste+edit the implementations of the `uint16_t` and `uint64_t` versions of `read_ntoh` because you found it was infeasible (i.e. not allowed by the compiler) to make the `int16_t` and `int64_t` versions just call the `uint16_t` and `uint64_t` versions?

Reply:
there is no `int64_t `implementation that is the `uint64_t `overloaded function, for the `int16_t  ` version, ya the compiler/clang-tidy complained if I called uint16_t version by static_cast